SwiftReflector/NewClassCompiler.cs

```diff
@@ -5443,33 +5452,40 @@ static string LocateFile (IEnumerable<string> paths, string modName)
 		List<ModuleDeclaration> GetModuleDeclarations (List<string> moduleDirectories, List<string> moduleNames,
 		                                               string outputDirectory, bool retainReflectedXmlOutput,
-		                                               List<string> targets, ErrorHandling errors)
+		                                               List<string> targets, ErrorHandling errors, bool isLibrary = false)
 		{
 			try {
 				string bestTarget = ChooseBestTarget (targets);
 
-				using (TempDirectoryFilenameProvider provider = new TempDirectoryFilenameProvider (null, true)) {
-					var targetInfo = ReflectorLocations.GetTargetInfo (bestTarget);
-					using (CustomSwiftCompiler compiler = new CustomSwiftCompiler (targetInfo, provider, false)) {
-						compiler.Verbose = Verbose;
-
-						var decls = compiler.ReflectToModules (
-							moduleDirectories.ToArray (), moduleDirectories.ToArray (),
-							"-framework XamGlue", moduleNames.ToArray ());
-
-						foreach (var mdecl in decls) {
-							if (!mdecl.IsCompilerCompatibleWith (CompilerVersion)) {
-								throw ErrorHelper.CreateError (ReflectorError.kCantHappenBase + 49, $"The module {mdecl.Name} was compiled with the swift compiler version {mdecl.SwiftCompilerVersion}. It is incompatible with the compiler in {SwiftCompilerLocations.SwiftCompilerBin} which is version {CompilerVersion}");
+				// TJ - if we have a dylib, we will be using our already generated xml
+				if (isLibrary) {
+					var decls = Reflector.FromXmlFile ("../../../DylibBinder/Modules/modules_tester.xml");
+					return decls;
+				} else {
+					using (TempDirectoryFilenameProvider provider = new TempDirectoryFilenameProvider (null, true)) {
+						var targetInfo = ReflectorLocations.GetTargetInfo (bestTarget);
+						using (CustomSwiftCompiler compiler = new CustomSwiftCompiler (targetInfo, provider, false)) {
+							compiler.Verbose = Verbose;
+
+							var decls = compiler.ReflectToModules (
+								moduleDirectories.ToArray (), moduleDirectories.ToArray (),
+								"-framework XamGlue", moduleNames.ToArray ());
+
+							foreach (var mdecl in decls) {
+								if (!mdecl.IsCompilerCompatibleWith (CompilerVersion)) {
+									throw ErrorHelper.CreateError (ReflectorError.kCantHappenBase + 49, $"The module {mdecl.Name} was compiled with the swift compiler version {mdecl.SwiftCompilerVersion}. It is incompatible with the compiler in {SwiftCompilerLocations.SwiftCompilerBin} which is version {CompilerVersion}");
+								}
 							}
-						}
 
-						if (retainReflectedXmlOutput) {
-							var files = Directory.GetFiles (provider.DirectoryPath).Where (fileName => fileName.EndsWith (".xml", StringComparison.OrdinalIgnoreCase)).ToList ();
-							CopyXmlOutput (outputDirectory, files);
+							if (retainReflectedXmlOutput) {
+								var files = Directory.GetFiles (provider.DirectoryPath).Where (fileName => fileName.EndsWith (".xml", StringComparison.OrdinalIgnoreCase)).ToList ();
```

Comment:
This look like you should be using a BCL method: https://docs.microsoft.com/en-us/dotnet/api/system.io.directory.getfiles?view=net-5.0#System_IO_Directory_GetFiles_System_String_System_String_System_IO_EnumerationOptions_

Reply:
This part was already written, I just added an if/else before it moving it over and GitHub thinking I am adding it. @stephen-hawley